### PR TITLE
Add support for external-traffic LocalOnly services

### PR DESCRIPTION
**Pull request by databus23**
cc @BugRoger

### CODE REVIEW

File: pkg/controller/externalservices.go [hunk 1]
```diff
@@ -210,22 +210,60 @@ func (c *ExternalServicesController) reconcile() error {
 			}
 		}
 
-		if _, ok, _ := c.endpoints.Get(route.Service); !ok {
+		if eps, ok, _ := c.endpoints.Get(route.Service); !ok {
 			if err := c.routes.Delete(route); err != nil {
 				return err
 			}
+		} else if hasLocalOnlyAnnotation(route.Service) {
```
Comment (BugRoger) [inline pkg/controller/externalservices.go:217]: This reads "If there is endpoints for this service AND the service has the local annotation AND there is NO endpoint on this node THEN delete the route". Hard to understand but probably right... 😄

**Review by BugRoger — APPROVED** [review]
LGTM in theory. In practice we'll have to see how nice the BGP converge times play along with deployments and endpoint changes.

**Comment by BugRoger** [comment]
@databus23 Merge at will. Doesn't hurt to have this in.

**Comment by databus23** [comment]
Yay!

**Pull request merged**